File: tests/data/program_analysis/strings/str05.f
C     File: str05.f
C     This program tests the INDEX() function on strings.
C     The output generated is:
C
C       2    0   6   3   0

      program main
      character(len = 10) str1
      integer n1, n2, n3, n4, n5

      str1 = "abcdef"

      n1 = index(str1, "bc")
      n2 = index(str1, "xyz")
      n3 = index(str1, "f ", back=.true.)
      n4 = index(str1, "cde", back=.true.)
      n5 = index(str1, "xyz", back=.true.)

      write (*, 10) n1, n2, n3, n4, n5

 10   format(5(I3,X))

      stop
      end program main
      

File: tests/data/program_analysis/strings/Translation/str05.py
from delphi.translators.for2py.format import *
from delphi.translators.for2py.strings import *

def main():
    str1 = String(10)

    str1.set_("abcdef")

    n1 = str1.f_index("bc")
    n2 = str1.f_index("xyz")
    n3 = str1.f_index("f ", ["back"])
    n4 = str1.f_index("cde", ["back"])
    n5 = str1.f_index("xyz", ["back"])

    fmt_10 = Format(['5(I3,X)'])
    write_str = fmt_10.write_line([n1, n2, n3, n4, n5])
    sys.stdout.write(write_str)

main()
